### PR TITLE
Improve eigenvals docs

### PR DESCRIPTION
**Pull request by sylee957**
#### Brief description of what is fixed or changed

I had corrected the wrong boldcases, by correctly formatting the doc.
And had added some details about undocumented key `multiple`.

And also had made the doc richer by adding some sections, and reformatted some contents.
These are before and after

image

image

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by moorepants** [comment]
These docs changes are great! Thanks for the effort.

I'd like to suggest that that the opening statement and the parameter/returns sections have no LaTeX formatted math so that they are readable via plain text interpreters. I think that the Notes section (below params/returns) should contain any latex formatted math to expand on things and explain in detail. This has been discussed elsewhere and there may not be consensus. FYI

File: sympy/matrices/matrices.py [hunk 1]
```diff
@@ -1117,25 +1117,61 @@ def eigenvals(self, error_when_incomplete=True, **flags):
         """Return eigenvalues using the Berkowitz agorithm to compute
         the characteristic polynomial.
 
+        Eigenvalues of a matrix `A` can be computed by solving a matrix
+        equation `det(A - \lambda I) = 0`
```
Comment (jksuom) [inline sympy/matrices/matrices.py]: This could probably be `\det`.

**Comment by sylee957** [comment]
@moorepants 
Moving some details to notes can be a good idea.
However, I think the `Notes` section have no themes currently,
Though I don't know if it is intended.

Though I have made the initial setups, but had just suspended the section.
And I have captured a screenshot about how this `Notes` section would blend up with other elements.
Will it be okay?

image

File: sympy/matrices/matrices.py [hunk 2]
```diff
@@ -1120,22 +1120,55 @@ def eigenvals(self, error_when_incomplete=True, **flags):
...
+            the key ``multiple``.
+
+        Raises
+        ======
+
+        MatrixError
+            If not enough roots had got computed.
+
+        NonSquareMatrixError
+            If attempted to compute eigenvalues from a non-sqaure matrix.
```
Comment (Abdullahjavednesar) [inline sympy/matrices/matrices.py]: `non-sqaure` -> `non-square`

**Comment by moorepants** [comment]
>  Will it be okay?

Looks great. My request was more-or-less a reminder to make sure the docstrings work reasonably well for command line uses as well as rendered uses.

File: sympy/matrices/matrices.py [hunk 3]
```diff
@@ -1120,22 +1120,55 @@ def eigenvals(self, error_when_incomplete=True, **flags):
         Parameters
         ==========
 
-        error_when_incomplete : bool
-            Raise an error when not all eigenvalues are computed. This is
-            caused by ``roots`` not returning a full list of eigenvalues.
+        error_when_incomplete : bool, optional
+            If it is set to ``True``, it will attempt to raise an error when
+            not every eigenvalues had got computed. This is caused by ``roots`` not
```
Comment (sidhantnagpal) [inline sympy/matrices/matrices.py]: ...it will raise an error if not all eigenvalues are computed.

File: sympy/matrices/matrices.py [hunk 4]
```diff
@@ -1120,22 +1120,55 @@ def eigenvals(self, error_when_incomplete=True, **flags):
...
+            If not enough roots had got computed.
+
+        NonSquareMatrixError
+            If attempted to compute eigenvalues from a non-sqaure matrix.
+
+        See Also
+        ========
+
+        charpoly
+        eigenvects
```
Comment (sidhantnagpal) [inline sympy/matrices/matrices.py]: I suppose these could be comma separated. Is there a reason for using newline?

**Comment by jksuom** [comment]
> invalid escape sequence \d by some procedure,

It is usually best to make docstrings containing latex code raw: `r"""...."""`.

**Comment by sidhantnagpal** [comment]
> almost every docs in matrices.py uses newlines for separating

Though this is a minor issue and they render similarly, I would like to point out that newlines are generally used when modules are included as [prefixes](https://github.com/sympy/sympy/blob/master/sympy/functions/elementary/complexes.py#L799). Also, using commas and single space reduces the file size and the number of lines in the file.

**Comment by sylee957** [comment]
@sidhantnagpal 
Actually, there was a problem that the `charpoly` link was broken.
I think adding a directive to `MatrixDeterminant` like b3e8e49cc4dfed9e92ed72522053c0245adf75fe and 168936beded25795c7e4eb8c0bb7e5ffb7c2d422 would be a better solution for this.

@jksuom 
I have changed to raw docstring

**Pull request merged**